sklearn/inspection/_permutation_importance.py

```diff
@@ -143,14 +159,36 @@ def permutation_importance(estimator, X, y, *, scoring=None, n_repeats=5,
     random_state = check_random_state(random_state)
     random_seed = random_state.randint(np.iinfo(np.int32).max + 1)
 
-    scorer = check_scoring(estimator, scoring=scoring)
-    baseline_score = _weights_scorer(scorer, estimator, X, y, sample_weight)
-
-    scores = Parallel(n_jobs=n_jobs)(delayed(_calculate_permutation_scores)(
-        estimator, X, y, sample_weight, col_idx, random_seed, n_repeats, scorer
-    ) for col_idx in range(X.shape[1]))
-
-    importances = baseline_score - np.array(scores)
-    return Bunch(importances_mean=np.mean(importances, axis=1),
-                 importances_std=np.std(importances, axis=1),
-                 importances=importances)
+    if scoring is None or isinstance(scoring, str) or callable(scoring):
+        scorer = check_scoring(estimator, scoring=scoring)
+        baseline_score = _weights_scorer(scorer, estimator, X, y,
+                                         sample_weight)
+
+        scores = Parallel(n_jobs=n_jobs)(
+            delayed(_calculate_permutation_scores)(
+                estimator, X, y, sample_weight, col_idx, random_seed,
+                n_repeats, scorer
+            ) for col_idx in range(X.shape[1]))
+
+        importances = baseline_score - np.array(scores)
+        return Bunch(importances_mean=np.mean(importances, axis=1),
+                     importances_std=np.std(importances, axis=1),
+                     importances=importances)
+    else:
+        scorers = _check_multimetric_scoring(estimator, scoring)
```

Comment:
I think that we need to create scorers using `scorers = _MultimetricScorer(scorers)` since the caching mechanism is within the class. `_check_multimetric_scoring` will only return a dict of validated scorer without cache implementation.

I think that the solution here is to make `_weight_scorer` accepting `_MultimetricScorer`. Currently, the code should not change, only the returned output will be slightly different (multiple score instead of a single score).

Reply:
I imagine that the outline of the code would look like:

```python
if scoring is None or isinstance(scoring, str) or callable(scoring):
    scorer = check_scoring(estimator, scoring=scoring)
else:
    scorers_dict = _check_multimetric_scoring(estimator, scoring)
    scorer = _MultimetricScoring(**scorers_dict)

    baseline_score = _weights_scorer(
        scorer, estimator, X, y, sample_weight=sample_weight)
    scores = Parallel(n_jobs=n_jobs)(
                delayed(_calculate_permutation_scores)(
                    estimator, X, y, sample_weight, col_idx, random_seed,
                    n_repeats, scorer
                ) for col_idx in range(X.shape[1]))

    if isinstance(baseline_score, dict):
         # unpack the scores since they come from multimetric scorer
    else:
        # a substraction is enough

     return importances 
```

Reply:
yes, more or less the implementation done so far. I'll let you check.